app/services/record_declarations/base.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+require "json_schema/validate_body_against_schema"
+
+module RecordDeclarations
+  class Base
+    attr_accessor :params
+    delegate :user_profile, :actual_lead_provider, to: :not_implemented_error
+
+    class << self
+      def call(params)
+        new(params).call
+      end
+
+      def required_params
+        %i[participant_id cpd_lead_provider declaration_type declaration_date course_identifier raw_event]
+      end
+    end
+
+    def not_implemented_error
+      raise NotImplementedError, "Method must be implemented"
+    end
+
+    def call
+      validate_schema!
+      add_participant_profile_params!
+
+      declaration = create_record!
+      validate_provider!
```

Comment:
Why does this happen after `create_record!`?

Reply:
Cause we want to store it if the providers are shadily trying to declare participants which don't belong to them. I think.